Asset Uploader UI Component/Code/Asset Uploader/Objective-C/AssetUploader/ViewController.m

```diff
@@ -261,7 +278,44 @@ - (void)syncFinished
     [[AdobeLibraryManager sharedInstance] deregisterDelegate:self];
 }
 
-#pragma mark - Priavte methods
+#pragma mark - Notification handlers
+
+- (void)authDidLogout:(NSNotification *)notification
+{
+    if (self.rootLibDir.length > 0)
+    {
+        [AdobeLibraryManager removeLocalLibraryFilesInRootFolder:self.rootLibDir withError:nil];
+        
+        // Clean up the root folder.
+        NSFileManager *fm = [NSFileManager defaultManager];
+        [fm removeItemAtPath:[[self.rootLibDir stringByDeletingLastPathComponent] stringByDeletingLastPathComponent] error:nil];
+        self.rootLibDir = nil;
+        [[NSUserDefaults standardUserDefaults] removeObjectForKey:@"userSpecificPathKey"];
+        [[NSUserDefaults standardUserDefaults] synchronize];
+    }
+}
+
+#pragma mark - Private methods
+
+- (NSString *)UUIDForUserSpecificPath
+{
+    // If we have a UUID then use it.
+    [[NSUserDefaults standardUserDefaults] synchronize];
+    NSString *userSpecificID = [[NSUserDefaults standardUserDefaults] objectForKey:@"userSpecificPathKey"];
+    
+    if (userSpecificID.length > 0)
+    {
+        return userSpecificID;
+    }
+    
+    // Generate a new UUID
+    userSpecificID = [[NSUUID UUID] UUIDString];
```

Comment:
Minor nitpick: `UUIDString` is a property so it should be accessed with a dot property.